pkg/scheduler/plugins/util/k8s/snapshot_test.go

```diff
@@ -0,0 +1,213 @@
+			},
+			Spec: v1.PodSpec{
+				Containers: []v1.Container{
+					{
+						Resources: v1.ResourceRequirements{
+							Requests: v1.ResourceList{
+								v1.ResourceCPU:    resource.MustParse("200m"),
+								v1.ResourceMemory: resource.MustParse("1Ki"),
+							},
+						},
+						Ports: []v1.ContainerPort{
+							{
+								HostIP:   "127.0.0.1",
+								HostPort: 8080,
+								Protocol: "TCP",
+							},
+						},
+					},
+				},
+				NodeName: nodeName,
+			},
+		}
+	)
+
+	tests := []struct {
+		name              string
+		nodeInfoMap       map[string]*framework.NodeInfo
+		expectedNodeInfos []*framework.NodeInfo
+		expectedPods      []*v1.Pod
+		expectErr         error
+	}{
+		{
+			name: "test snapshot operation",
+			nodeInfoMap: map[string]*framework.NodeInfo{nodeName: {
+				Requested:        &framework.Resource{},
+				NonZeroRequested: &framework.Resource{},
+				Allocatable:      &framework.Resource{},
+				Generation:       2,
+				UsedPorts: framework.HostPortInfo{
+					"127.0.0.1": map[framework.ProtocolPort]struct{}{
+						{Protocol: "TCP", Port: 80}:   {},
+						{Protocol: "TCP", Port: 8080}: {},
+					},
+				},
+				ImageStates:  map[string]*framework.ImageStateSummary{},
+				PVCRefCounts: map[string]int{},
+				Pods: []*framework.PodInfo{
+					{
+						Pod: pod1,
+					},
+					{
+						Pod: pod2,
+					},
+				},
+				PodsWithAffinity: []*framework.PodInfo{
+					{
+						Pod: pod1,
+					},
+					{
+						Pod: pod2,
+					},
+				},
+				PodsWithRequiredAntiAffinity: []*framework.PodInfo{
+					{
+						Pod: pod1,
+					},
+					{
+						Pod: pod2,
+					},
+				},
+			}},
+			expectedNodeInfos: []*framework.NodeInfo{{
+				Requested:        &framework.Resource{},
+				NonZeroRequested: &framework.Resource{},
+				Allocatable:      &framework.Resource{},
+				Generation:       2,
+				UsedPorts: framework.HostPortInfo{
+					"127.0.0.1": map[framework.ProtocolPort]struct{}{
+						{Protocol: "TCP", Port: 80}:   {},
+						{Protocol: "TCP", Port: 8080}: {},
+					},
+				},
+				ImageStates:  map[string]*framework.ImageStateSummary{},
+				PVCRefCounts: map[string]int{},
+				Pods: []*framework.PodInfo{
+					{
+						Pod: pod1,
+					},
+					{
+						Pod: pod2,
+					},
+				},
+				PodsWithAffinity: []*framework.PodInfo{
+					{
+						Pod: pod1,
+					},
+					{
+						Pod: pod2,
+					},
+				},
+				PodsWithRequiredAntiAffinity: []*framework.PodInfo{
+					{
+						Pod: pod1,
+					},
+					{
+						Pod: pod2,
+					},
+				},
+			}},
+			expectedPods: []*v1.Pod{pod2},
+			expectErr:    fmt.Errorf("nodeinfo not found for node name %q", nodeName),
+		},
+	}
+
+	for _, tc := range tests {
+		t.Run(tc.name, func(t *testing.T) {
+			snapshot := NewSnapshot(tc.nodeInfoMap)
+			nodeInfoList, err := snapshot.List()
+			if !reflect.DeepEqual(nodeInfoList, tc.expectedNodeInfos) || err != nil {
+				t.Errorf("unexpected list nodeInfos value (+got: %s/-want: %s), err: %s", tc.expectedNodeInfos, nodeInfoList, err)
```

Comment:
use t.fail. same as the others

Reply:
could i use `t.Fatalf`,  `t.Fatalf` can print err log, but `t.Fail()` can't.